### PR TITLE
feat: Upgrade for relay-runtime v8.x

### PR DESCRIPTION
**Pull request by taion**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/facebook/relay/commit/edda8aa94e204df0a361cfa495ad5a9ed1db4c31, https://github.com/facebook/relay/commit/5febecd25604e15cb14fe918755ae70ef90b261c
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/relay-runtime/lib/handlers/getRelayDefaultMissingFieldHandlers.d.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import { MissingFieldHandler } from '../store/RelayStoreTypes';
+
+export default function getHandler(): MissingFieldHandler[];
```
Comment (taion) [inline types/relay-runtime/lib/handlers/getRelayDefaultMissingFieldHandlers.d.ts]: yes, it's named `getHandler` in https://github.com/facebook/relay/blob/3c7a75666ef3d0cc9d47d1f86ecffb744bd8f020/packages/relay-runtime/handlers/getRelayDefaultMissingFieldHandlers.js#L22 ¯\\\_(ツ)_/¯

File: types/relay-runtime/tslint.json [hunk 2]
```diff
@@ -2,6 +2,8 @@
     "extends": "dtslint/dt.json",
     "rules": {
         // We have some internal types that are helpfully tagged with @private
-        "no-redundant-jsdoc-2": false
+        "no-redundant-jsdoc-2": false,
+        "no-unnecessary-generics": false,
```
Comment (taion) [inline types/relay-runtime/tslint.json]: this and the below reflected existing failures; not sure how they didn't come up before

Comment (alloy) [reply types/relay-runtime/tslint.json]: Perhaps an older version of tslint was used previously?  🤷‍♂

**Comment by alloy** [comment]
LGTM 👌

**Pull request merged**